lib/src/fnc/string.rs

```diff
@@ -7,16 +7,18 @@ pub fn concat(_: &Context, args: Vec<Value>) -> Result<Value, Error> {
 	Ok(args.into_iter().map(|x| x.as_string()).collect::<Vec<_>>().concat().into())
 }
 
-pub fn ends_with(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
-	let val = args.remove(0).as_string();
-	let chr = args.remove(0).as_string();
+pub fn ends_with(_: &Context, args: Vec<Value>) -> Result<Value, Error> {
+	let args: [Value; 2] = args.try_into().unwrap();
+	let [val, chr] = args.map(Value::as_string);
 	Ok(val.ends_with(&chr).into())
 }
 
-pub fn join(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
-	let chr = args.remove(0).as_string();
-	let val = args.into_iter().map(|x| x.as_string());
-	let val = val.collect::<Vec<_>>().join(&chr);
+pub fn join(_: &Context, args: Vec<Value>) -> Result<Value, Error> {
+	let mut args = args.into_iter().map(Value::as_string);
+	let chr = args.next().unwrap();
+	// FIXME: Use intersperse to avoid intermediate allocation once stable
+	// https://github.com/rust-lang/rust/issues/79524
+	let val = args.collect::<Vec<_>>().join(&chr);
```

Comment:
remove inefficient removal of first index from vec, and note that, in the future, the `collect` won't be necessary